### PR TITLE
android: Emulation loading UI and fixes

### PR DESCRIPTION
**Pull request by t895**
- I added custom loaders for coil to generalize the game cover loading process. This also makes async cover loading a lot more consistent and a little faster.
- There's new loading animations for startup and shutdown. This eliminates the shader loading dialog and combines it into prettier UI
startup
shuttingdown
- Also a bit of a silly fix. Checking to see if emulation was running would rely on accessing the mutex also used for starting/stopping emulation. This would result in a long stall on the UI thread and often result in a crash if it took longer than the foreground service timeout (5 seconds). Instead of waiting for the mutex to be unlocked, now we just use atomic booleans to track emulation state (only for what UI/input/perf stats care about).

### CODE REVIEW

**Comment by t895** [comment]
cc @Schplee for UI review

**Comment by t895** [comment]
Just noticed that the loading text was broken (you could see the progress and max text were always the same in the gif). Just pushed a fix.

**Comment by t895** [comment]
Just altered the design a bit. I removed the # of shaders processed text to prevent weird layout scaling issues at different dpis.

File: src/android/app/src/main/java/org/yuzu/yuzu_emu/fragments/EmulationFragment.kt [hunk 1]
```diff
@@ -281,17 +340,15 @@ class EmulationFragment : Fragment(), SurfaceHolder.Callback {
             val FRAMETIME = 2
             val SPEED = 3
             perfStatsUpdater = {
-                val perfStats = NativeLibrary.getPerfStats()
-                if (perfStats[FPS] > 0 && _binding != null) {
-                    binding.showFpsText.text = String.format("FPS: %.1f", perfStats[FPS])
-                }
-
-                if (!emulationState.isStopped) {
+                if (emulationViewModel.emulationStarted.value == true) {
```
Comment (FearlessTobi) [inline src/android/app/src/main/java/org/yuzu/yuzu_emu/fragments/EmulationFragment.kt:343]:
```suggestion
                if (emulationViewModel.emulationStarted.value) {
```

File: src/android/app/src/main/java/org/yuzu/yuzu_emu/model/EmulationViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    private val _shaderProgress = MutableLiveData(0)
+    val shaderProgress: LiveData<Int> get() = _shaderProgress
+
+    private val _totalShaders = MutableLiveData(0)
+    val totalShaders: LiveData<Int> get() = _totalShaders
+
+    private val _shaderMessage = MutableLiveData("")
+    val shaderMessage: LiveData<String> get() = _shaderMessage
+
+    fun setEmulationStarted(started: Boolean) {
```
Comment (FearlessTobi) [inline src/android/app/src/main/java/org/yuzu/yuzu_emu/model/EmulationViewModel.kt:26]: i think there is an easier way to do this in kotlin, but i don't remember

File: src/android/app/src/main/java/org/yuzu/yuzu_emu/utils/GameIconUtils.kt [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+import org.yuzu.yuzu_emu.YuzuApplication
+import org.yuzu.yuzu_emu.model.Game
+
+class GameIconFetcher(
+    private val game: Game,
+    private val options: Options
+) : Fetcher {
+    override suspend fun fetch(): FetchResult {
+        return DrawableResult(
+            drawable = decodeGameIcon(game.path)!!.toDrawable(options.context.resources),
```
Comment (FearlessTobi) [inline src/android/app/src/main/java/org/yuzu/yuzu_emu/utils/GameIconUtils.kt:30]: should we handle the error here if decoding the bitmap fails?

File: src/android/app/src/main/res/values-de/strings.xml [hunk 4]
```diff
@@ -209,7 +209,6 @@
     <string name="emulation_pause">Emulation pausieren</string>
     <string name="emulation_unpause">Emulation fortsetzen</string>
     <string name="emulation_input_overlay">Overlay-Optionen</string>
-    <string name="emulation_game_loading">Spiel lädt…</string>
```
Comment (FearlessTobi) [inline src/android/app/src/main/res/values-de/strings.xml:212]: this should probably be handled by transifex and not manually

**Review by FearlessTobi — APPROVED** [review]
LGTM overall

File: src/android/app/src/main/res/values-de/strings.xml (hunk 4 above)
Comment (t895) [reply src/android/app/src/main/res/values-de/strings.xml:212]: You get a compiler error if the source string is missing

File: src/android/app/src/main/java/org/yuzu/yuzu_emu/fragments/EmulationFragment.kt (hunk 1 above)
Comment (t895) [reply src/android/app/src/main/java/org/yuzu/yuzu_emu/fragments/EmulationFragment.kt:343]: 'value' is nullable and this is the null-safe way to check

File: src/android/app/src/main/java/org/yuzu/yuzu_emu/model/EmulationViewModel.kt (hunk 2 above)
Comment (t895) [reply src/android/app/src/main/java/org/yuzu/yuzu_emu/model/EmulationViewModel.kt:26]: I do want to migrate to State Flows but in a later PR

File: src/android/app/src/main/java/org/yuzu/yuzu_emu/utils/GameIconUtils.kt (hunk 3 above)
Comment (t895) [reply src/android/app/src/main/java/org/yuzu/yuzu_emu/utils/GameIconUtils.kt:30]: It's automatically handled by Coil and it just returns the error bitmap we provide

**Comment by liamwhite** [comment]
This is fixed in v72 (not yet published)

**Pull request merged**